core/lib/client-cookies.ts

```diff
@@ -0,0 +1,57 @@
+export const FORCE_REFRESH_COOKIE = 'force-refresh';
+
+interface ClientCookieOptions {
+  expires: Date;
+  path: string;
+  domain: string;
+  secure: boolean;
+  sameSite: 'Strict' | 'Lax' | 'None';
+  maxAge: number;
+}
+
+const cookiePropertyMap: Record<string, string> = {
+  expires: 'Expires',
+  path: 'Path',
+  domain: 'Domain',
+  secure: 'Secure',
+  sameSite: 'SameSite',
+  maxAge: 'Max-Age',
+};
+
+export function getCookieValue(name: string): string | null {
+  const cookies = document.cookie.split('; ');
+  const cookie = cookies.find((c) => c.startsWith(`${name}=`));
+
+  if (!cookie) {
+    return null;
+  }
+
+  const cookieValue = cookie.split('=')[1];
+
+  return cookieValue ?? null;
+}
+
+export function setCookie(
+  name: string,
+  value: string,
+  options?: Partial<ClientCookieOptions>,
+): void {
+  const opts = options ?? {};
+  const cookieOptions = Object.entries(opts)
+    .reduce<string[]>((acc, [key, optValue]) => {
+      const propKey = cookiePropertyMap[key];
+
+      if (!propKey) {
+        return acc;
+      }
+
+      if (propKey === 'Secure' && optValue === true) {
+        return [...acc, propKey];
+      }
```

Comment:
Can you explain why we're doing this? 🙏

Reply:
@jorgemoya It's just a simple client-side cookie function, when setting a cookie on the client, the `Secure` flag has no value so `{ secure: true }` becomes `my_cookie=value; Secure` and `{ secure: false }` would be `my_cookie=value;`

Reply:
A better example:
```
setCookie('my_cookie', 'test_value', { secure: true, maxAge: 1000 })
```

would become:
```
my_cookie=test_value; Secure; Max-Age=1000;`
```